skxray/diff_roi_choice.py

```diff
@@ -360,18 +383,85 @@ def _process_rings(num_rings, img_dim, ring_vals, ring_inds):
         pixel indices for the required rings
 
     """
-    # find the pixel list
-    w = np.where(ring_inds > 0)
-    grid = np.indices((img_dim[0], img_dim[1]))
-    pixel_list = (grid[0]*img_dim[1] + grid[1]).flatten()[w]
+    mesh = np.zeros(detector_size, dtype=np.int64)
+    angles = np.linspace(0, 360, num_angles)
+
+    yy, xx = np.mgrid[:detector_size[0], :detector_size[1]]
+    y_ = (np.flipud(yy) - calibrated_center[1])
+    x_ = (xx - calibrated_center[0])
+    grid_values = np.float_(np.hypot(x_, y_))
+    angle_grid = np.arctan2(y_, x_)*180/(np.pi)
+
+    for i in range(num_angles-1):
+        if angles[i + 1] <= 180:
+            vl = ((angles[i] < angle_grid) &
+                  (angle_grid <= angles[i + 1]))
+
+        elif (angles[i] < 180) & (angles[i + 1] > 180):
+            vl = ((angles[i] < angle_grid) & (angle_grid <= 180) |
+                  (-180 < angle_grid) &
+                  (angle_grid <= angles[i + 1] - 360))
+
+        elif (angles[i] == 180):
+            vl = ((-180 < angle_grid) &
+                  (angle_grid <= angles[i + 1] - 360))
+
+        else:
+            vl = ((angles[i] - 360 < angle_grid) &
+                  (angle_grid <= angles[i + 1] - 360))
+
+        mesh[vl] = i + 1
 
-    ring_inds = ring_inds[ring_inds > 0]
+    mesh[grid_values > radius] = 0
 
-    ring_vals = np.array(ring_vals)
-    ring_vals = ring_vals.reshape(num_rings, 2)
+    roi_inds, num_pixels, pixel_list = _process_rois(np.ravel(mesh),
+                                                     detector_size,
+                                                     num_angles)
+    return roi_inds, num_pixels, pixel_list
+
+
+def _process_rois(mesh, detector_size, num_rois):
+    """
+    This is a helper function to find the indices of the required rois,
+    and count the number of pixels in each rois, and pixels list for
+    the required rois.
+
+    Parameters
+    ----------
+    mesh : ndarray
+        indices of the required roi's
+        ([detector_size[0]*detector_size[1]], )
+
+    detector_size : tuple
+        2 element tuple defining the number of pixels
+        in the detector.
+        Order is (num_rows, num_columns)
 
-    # number of pixels in each ring
-    num_pixels = np.bincount(ring_inds, minlength=(num_rings+1))
+    num_rois: int
+        number of region of interests
+
+    Returns
+    -------
+    roi_inds : ndarray
+        indices of the required roi's
+        (after discarding zero values from the shape
+        ([detector_size[0]*detector_size[1]], )
+
+    num_pixels : ndarray
+        number of pixels in each roi's
```

Comment:
This information can be calculated from roi_inds. Do you still want to return this again?